File: tests/pages/contribute/friends.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.contribute.base import ContributeBasePage


class ContributeFriendsPage(ContributeBasePage):

    _url = '{base_url}/{locale}/contribute/friends'

    _show_signup_form_button_locator = (By.ID, 'ff-show-signup-form')
    _signup_form_locator = (By.ID, 'newsletter-form')

    @property
    def is_signup_form_displayed(self):
        return self.is_element_displayed(self._signup_form_locator)

    def click_show_signup_form(self):
        self.find_element(self._show_signup_form_button_locator).click()
        self.wait.until(lambda s: self.is_signup_form_displayed)

File: media/js/mozorg/contribute/friends.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function($, Mozilla) {
    'use strict';

    var $ffShowSignupForm = $('#ff-show-signup-form');
    var $ffSignupArea = $('#ff-signup');

    $ffShowSignupForm.on('click', function() {
        $ffShowSignupForm.addClass('invisible').off('click');

        // slide the form container down
        $ffSignupArea.slideDown('normal', function() {
            $('#id_email').focus();

            // scroll to top of revealed form
            Mozilla.smoothScroll({
                top: $ffSignupArea.offset().top - 60
            });
        });

        window.dataLayer.push({
            'event': 'firefox-sync-interaction',
            'interaction': 'display signup form'
        });
    });

    Mozilla.SVGImage.fallback();

    /*
    Unsure if this is necessary - think we already have tracking on this form?
    Waiting on comment from garethc in the bug.

    $('#footer_email_submit').on('click', function() {
        window.dataLayer.push({
            'event': 'firefox-sync-interaction',
            'interaction': 'signup form submit'
        });
    });
    */
})(window.jQuery, window.Mozilla);
